onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/OperatorRegistration.cpp

```diff
@@ -220,7 +222,7 @@ const static SupportedTensorDataTypes supportedTypeListAllScalars[1] = { Support
 const static SupportedTensorDataTypes supportedTypeListBool[1] = {SupportedTensorDataTypes::Bool};
 const static SupportedTensorDataTypes supportedTypeListTopK[2] = {SupportedTensorDataTypes::Float16to32, SupportedTensorDataTypes::Int64};
 const static SupportedTensorDataTypes supportedTypeListIndices[1] = { SupportedTensorDataTypes::Int32|SupportedTensorDataTypes::Int64 };
-const static SupportedTensorDataTypes supportedTypeListCast[2] = { SupportedTensorDataTypes::Scalars8to32, SupportedTensorDataTypes::Scalars8to32 };
+const static SupportedTensorDataTypes supportedTypeListCast[2] = { SupportedTensorDataTypes::AllScalars | SupportedTensorDataTypes::Bool, SupportedTensorDataTypes::AllScalars | SupportedTensorDataTypes::Bool };
```

Comment:
Re-enabling 64-bit support before we *truly* support 64-bit will regress one of our models (I forget which one). See the history for Cast for the bug Shrinath fixed where -1 was being truncated in Cast due to our 32-bit 2x stride hack.

Reply:
Good point.  It's a shame that ArgMax emits Int64.  This should probably wait until int64 is properly supported, and/or we have graph logic to track where the existing hack is safe.